### ISSUE TITLE
Add MAXIM model

### ISSUE THREAD

**Issue by innat** [keras-team/keras-cv#924]
**Short Description**

**MAXIM: Multi-Axis MLP for Image Processing**: I think it's a follow-up work of MaxViT from google. It shows a great performance on the following low-vision task, i.e. for OCR.

![image3](https://user-images.githubusercontent.com/17668390/195910903-e6bfe5e0-35da-4323-ac84-8720acda2d30.png)

It looks useful for real world problem (blurry, haze, rain-drop). Such problems are very acute for vision project. So, such solutions might weight quick adaptation of kcv in industrial projects ([alike](https://github.com/keras-team/keras-cv/issues/532)).

**Papers**

- https://arxiv.org/pdf/2201.02973.pdf
![image](https://user-images.githubusercontent.com/17668390/195911106-59fe4b8f-995e-45d2-807a-cbab6d8aaad1.png)

**Existing Implementations**

- In **Jax**: https://github.com/google-research/maxim
- [maxim-tf](https://github.com/sayakpaul/maxim-tf)
- Google-blog: https://ai.googleblog.com/2022/09/a-multi-axis-approach-for-vision.html
- pytorch: https://github.com/vztu/maxim-pytorch (WIP).

cc. @vztu

**Comment by tanzhenyu** [comment]
Open this up for contribution

**Comment by innat** [comment]
cc @sayakpaul

**Comment by IMvision12** [comment]
I'm working on integrating the maxim model and referencing the original flax implementation. I have one question: do we need weights as well?

**Comment by tanzhenyu** [comment]
> I'm working on integrating the maxim model and referencing the original flax implementation. I have one question: do we need weights as well? @tanzhenyu

Yes, we need the weights. The development cycle would be similar to ViT: https://github.com/keras-team/keras-cv/pull/1014, add the model and port the weights, also add the conversion scripts for porting the weights

**Comment by bhack** [comment]
> Yes, we need the weights. The development cycle would be similar to ViT: https://github.com/keras-team/keras-cv/pull/1014, add the model and port the weights, also add the conversion scripts for porting the weights

I hope we could standardize a bit the model  contribution process after we will handle:

https://github.com/keras-team/keras-cv/discussions/954

**Comment by IMvision12** [comment]
> > I'm working on integrating the maxim model and referencing the original flax implementation. I have one question: do we need weights as well? @tanzhenyu
> 
> Yes, we need the weights. The development cycle would be similar to ViT: #1014, add the model and port the weights, also add the conversion scripts for porting the weights

Okay will add the conversion script too!!

**Comment by innat** [comment]
@IMvision12 
Have you checked ?
https://github.com/sayakpaul/maxim-tf

**Comment by IMvision12** [comment]
> @IMvision12 Have you checked ? https://github.com/sayakpaul/maxim-tf

Oh we have tf implementation for maxim, because when I saw on [paper with code](https://paperswithcode.com/paper/maxim-multi-axis-mlp-for-image-processing) there was only official flax implementation

**Comment by IMvision12** [comment]
So, now we have weights for maxim here on tf [hub](https://tfhub.dev/sayakpaul/collections/maxim/1) uploaded by @sayakpaul, so do we still require a conversion script? @tanzhenyu

**Comment by sachinprasadhs** [comment]
Thanks for reporting the issue! We have consolidated the development of KerasCV into the new KerasHub package, which supports image, text, and multi-modal models. Please read https://github.com/keras-team/keras-hub/issues/1831. KerasHub will support all the core functionality of KerasCV.

KerasHub can be installed with !pip install -U keras-hub. Documentation and guides are available at [keras.io/keras_hub](https://keras.io/keras_hub/).

With our focus shifted to KerasHub, we are not planning any further development or releases in KerasCV. If you encounter a KerasCV feature that is missing from KerasHub, or would like to propose an addition to the library, please file [an issue with KerasHub](https://github.com/keras-team/keras-hub/issues).

**Comment by innat** [comment]
@sachinprasadhs 
> If you encounter a KerasCV feature that is missing from KerasHub, or would like to propose an addition to the library, please file [an issue with KerasHub](https://github.com/keras-team/keras-hub/issues).

Firstly, identifying which feature is required or missing can be done effectively, either by practitioners or the Keras team. Tickets in `keras-cv` were created only when such features were **NOT** available or when a specific need was encountered. Commenting after a significant delay and requesting the recreation of the same issue is neither efficient nor appropriate. Kindly move this ticket (and any other similar tickets with the same context) from `keras-cv` to `keras-hub`.
